SmartDeviceLink/SDLProxy.m

```diff
@@ -694,9 +693,11 @@ - (void)removeDelegate:(NSObject<SDLProxyListener> *)delegate {
 }
 
 - (void)invokeMethodOnDelegates:(SEL)aSelector withObject:(id)object {
+    __weak SDLProxy *weakSelf = self;
+    
     dispatch_async(dispatch_get_main_queue(), ^{
-        @autoreleasepool {
-            for (id<SDLProxyListener> listener in self.proxyListeners) {
+        if (!_alreadyDestructed){
```

Comment:
This should never be necessary, that whole process, including this variable, shouldn't exist. Have you seen errors based on this?

Why was `@autoreleasepool` removed? It should not have been.